### PR TITLE
Adds worms

### PR DESCRIPTION
**Pull request by Ketrai**
- adds worms
- adds three different types of worms.
- adds the ability to dig up worms from grass tiles.
- adds a cheap bait crate, consisting of 5 cans of sickly worms for 10 points.
- adds a deluxe bait crate, consisting of 8 cans, containing either deluxe or regular worms for 40 points.
- adds a fishing vendor, here you can buy bait and fishing equipment. Use sparingely.

Why exactly? Well, firstly to admit to the stereotype of fishing with worms.
Secondly to add bait items that aren't useful for eating, so that the station crew doesn't just go and plunder vendors to go fishing.

SPRITES:

fishingvendor2

### CODE REVIEW

File: code/modules/food/food/snacks.dm [hunk 1]
```diff
@@ -83,9 +84,23 @@
...
-			if (fullness > (550 * (1 + M.overeatduration / 2000)))	// The more you eat - the more you can eat
-				to_chat(M, "<span class='danger'>You cannot force any more of [src] to go down your throat.</span>")
+			if (fullness > 550 && fullness <= 650)
+				to_chat(M, "<span class='notice'>You swallow some more of the [src], causing your belly to swell out a little.</span>")
+			if (fullness > 650 && fullness <= 1000)
+				to_chat(M, "<span class='notice'>You stuff yourself with the [src]. Your stomach feels very heavy.</span>")
+			if (fullness > 1000 && fullness <= 3000)
+				to_chat(M, "<span class='notice'>You gluttonously swallow down the hunk of [src]. You're so gorged, it's hard to stand.</span>")
+			if (fullness > 3000 && fullness <= 5500)
+				to_chat(M, "<span class='danger'>You force the piece of [src] down your throat. You can feel your stomach getting firm as it reaches its limits.</span>")
```
Comment (mistyLuminescence) [inline code/modules/food/food/snacks.dm]: 👀 i feel like we should keep this stuff to be vore-specific

Comment (Ketrai) [reply code/modules/food/food/snacks.dm]: Oh uh, woops. I forgot to edit that out. I started coding this downstream. Figured what i'm trying to code isn't too specific for the codebase that I might as well code it here. Lemme fix this.

**Comment by Ketrai** [comment]
Can't figure out how to make it take a moment to dig a tile for worms yet. I figured I'd use do_after, but it doesn't like that.

**Comment by Ketrai** [comment]
Tested latest version, everything works.

**Pull request merged**